modules/xfd.js

```diff
@@ -0,0 +1,2052 @@
+            '.select2-results .select2-results__group { padding-top: 1px; padding-bottom: 1px; } ' +
+            // Adjust font size
+            '.select2-container .select2-dropdown .select2-results { font-size: 13px; }' +
+            '.select2-container .selection .select2-selection__rendered { font-size: 13px; }' +
+            // Make the tiny cross larger
+            '.select2-selection__choice__remove { font-size: 130%; }');
+    };
+    Afd.prototype.evaluate = function () {
+        var _this = this;
+        var tm = _super.prototype.evaluate.call(this);
+        tm.add(this.checkPage, []);
+        tm.add(this.determineDiscussionPage, []);
+        tm.add(this.createDiscussionPage, [this.checkPage, this.determineDiscussionPage]);
+        // create discussion page before linking or transcluding it from anywhere, so that
+        // there's no need to do any purging later (#364)
+        tm.add(this.tagPage, [this.checkPage, this.createDiscussionPage]); // tagPage has an arg coming from checkPage
+        tm.add(this.addToList, [this.createDiscussionPage]);
+        tm.add(this.addToDelsortLists, [this.createDiscussionPage]);
+        tm.add(this.patrolPage, [this.checkPage]);
+        tm.add(this.fetchCreatorInfo, []);
+        tm.add(this.notifyCreator, [this.createDiscussionPage, this.fetchCreatorInfo]);
+        tm.add(this.addToLog, [this.notifyCreator]);
+        tm.execute().then(function () { return _this.redirectToDiscussion(); });
+    };
+    Afd.prototype.preprocessParams = function () {
+        this.params.lookupNonRedirectCreator = true; // for this.fetchCreatorInfo()
+    };
+    Afd.prototype.determineDiscussionPage = function () {
+        var params = this.params;
+        return new Morebits.wiki.api('Determining discussion page', {
+            'action': 'query',
+            'list': 'allpages',
+            'apprefix': 'Articles for deletion/' + Morebits.pageNameNorm,
+            'apnamespace': 4,
+            'apfilterredir': 'nonredirects',
+            'aplimit': 'max' // 500 is max for normal users, 5000 for bots and sysops
+        }).post().then(function (apiobj) {
+            var xmlDoc = apiobj.responseXML;
+            var titles = $(xmlDoc).find('allpages p');
+            // There has been no earlier entries with this prefix, just go on.
+            if (titles.length <= 0) {
+                params.numbering = params.number = '';
+            }
+            else {
+                var number = 0;
+                for (var i = 0; i < titles.length; ++i) {
+                    var title = titles[i].getAttribute('title');
+                    // First, simple test, is there an instance with this exact name?
+                    if (title === 'Wikipedia:Articles for deletion/' + Morebits.pageNameNorm) {
+                        number = Math.max(number, 1);
+                        continue;
+                    }
+                    var order_re = new RegExp('^' +
```

Comment:
since the creation of a new RegExp object is likely a bit expensive, and since it doesn't change each time through the `for` loop, suggest creating once outside of the loop